src/lem/store.rs

```diff
@@ -64,6 +64,40 @@ pub struct Store<F: LurkField> {
     z_cache: FrozenMap<Ptr<F>, Box<ZPtr<F>>>,
 
     comms: FrozenMap<FWrap<F>, Box<(F, Ptr<F>)>>, // hash -> (secret, src)
+
+    pub hash3zeros: F,
+    pub hash4zeros: F,
+    pub hash6zeros: F,
+    pub hash8zeros: F,
+}
+
+impl<F: LurkField> Default for Store<F> {
+    fn default() -> Self {
+        let poseidon_cache = PoseidonCache::default();
+        let hash3zeros = poseidon_cache.hash3(&[F::ZERO; 3]);
+        let hash4zeros = poseidon_cache.hash4(&[F::ZERO; 4]);
+        let hash6zeros = poseidon_cache.hash6(&[F::ZERO; 6]);
+        let hash8zeros = poseidon_cache.hash8(&[F::ZERO; 8]);
```

Comment:
This is the extra cost of store startup

Reply:
Could we parallelize this?

Reply:
I think it can be parallelized, but I am not sure it's worth it since the `PoseidonCache` would be a locking bottleneck across threads

Reply:
The cache has four different locations per arity, writing to these locations only requires a shared reference thanks to the `FrozenMap`, and the locking is internal to each of these. What's the problem?

Reply:
You're right. But I tried it and the difference in performance wasn't clear enough to justify the increase in complexity:
```text
store_benchmark/store_go_base_bls12/_10_16                                                                           
                        time:   [152.22 µs 152.32 µs 152.40 µs]
                        change: [-2.7313% -2.6485% -2.5683%] (p = 0.00 < 0.05)
                        Performance has improved.
store_benchmark/store_go_base_pallas/_10_16                                                                           
                        time:   [154.93 µs 155.08 µs 155.24 µs]
                        change: [-1.0801% -0.7544% -0.2272%] (p = 0.00 < 0.05)
                        Change within noise threshold.
store_benchmark/store_go_base_bls12/_10_160                                                                           
                        time:   [158.65 µs 158.75 µs 158.86 µs]
                        change: [+1.2575% +1.3732% +1.4909%] (p = 0.00 < 0.05)
                        Performance has regressed.
store_benchmark/store_go_base_pallas/_10_160                                                                           
                        time:   [157.69 µs 158.22 µs 158.95 µs]
                        change: [+0.5752% +0.7856% +1.0481%] (p = 0.00 < 0.05)
                        Change within noise threshold.
```

This is how I did it:
```rust
let hashes = (0..4).into_par_iter().map(|i| {
            match i {
                0 => poseidon_cache.hash3(&[F::ZERO; 3]),
                1 => poseidon_cache.hash4(&[F::ZERO; 4]),
                2 => poseidon_cache.hash6(&[F::ZERO; 6]),
                3 => poseidon_cache.hash8(&[F::ZERO; 8]),
                _ => unreachable!(),
            }
        }).collect::<Vec<_>>();
```

```rust
            hash3zeros: hashes[0],
            hash4zeros: hashes[1],
            hash6zeros: hashes[2],
            hash8zeros: hashes[3],
```